### DISCUSSION TITLE
Feature request: USB barcode scanner support + global search [port from archived project]

### DISCUSSION THREAD

**Discussion by filp** [sysadminsmedia/homebox#740] [Ideas]
### What is the problem you are trying to solve with this feature?

- I want to be able to quickly search for items from any page in the application, without having to go to the search page first
- I'd like to use common consumer USB barcode scanners to quickly search for items

### What is the solution you are proposing?

I implemented this functionality in the now archived project, but it was never considered for merging upstream:

https://github.com/hay-kot/homebox/pull/658

If there is sufficient interest, I can update the PR to ensure it works with the latest version of the project.

### Contributions

- [X] I have searched through existing issues and feature requests to see if my idea has already been proposed.
- [X] If this feature is accepted, I would be willing to help implement and maintain this feature.
- [ ] If this feature is accepted, I'm willing to sponsor the development of this feature.

**Comment by tankerkiller125** [comment]
Please feel free to update the PR for this version :) we'd be happy to have a deeper look and potentially merge it.

**Comment by gab696** [comment]
I would find this useful too !

**Comment by tankerkiller125** [comment]
Please note we're converting this issue into a discussion as per our new procedures/policies. You can read more about that in our announcement #712
